### PR TITLE
prototype to create wallet

### PR DESCRIPTION
**Pull request by kzl5010**

### CODE REVIEW

File: app/src/api/orders/index.ts [hunk 1]
```diff
@@ -90,8 +90,27 @@ export const useUpdateOrder = () => {
...
+      await queryClient.invalidateQueries(["orders", id]);
     },
   });
 };
+
+export const useCreateSolanaWallet = () => {
+  const queryClient = useQueryClient();
+  const { session } = useAuth();
+
+  return useMutation({
```
Comment (kzl5010) [inline app/src/api/orders/index.ts:103]: hey Dan, i think this should be a working example of how the webapp would make requests to the web server. 

i haven't used expo web server so not totally sure how that works with auth middleware / adding users to the context but this *should* hit the createSolanaWallet endpoint. and then we should also addan invalidate query for whatever cache the get solana wallet  query ends up using.

File: app/src/components/EditScreenInfo.tsx [hunk 2]
```diff
@@ -1,12 +1,29 @@
...
+import { ExternalLink } from "./ExternalLink";
+import { MonoText } from "./StyledText";
+import { Text, View } from "./Themed";
+import { useAuth } from "@/providers/AuthProvider";
+import { useCreateSolanaWallet } from "@/api/orders";
 
 export default function EditScreenInfo({ path }: { path: string }) {
+  // example of an auth context consumer
+  const { session, loading, profile } = useAuth();
+  const { mutateAsync: createSolana } = useCreateSolanaWallet();
```
Comment (kzl5010) [inline app/src/components/EditScreenInfo.tsx:14]: and then here's an example of consuming the auth context in a component, in your routing component you'd want to redirect if they're not logged in. 

and then this should work for to create a solana wallet but i wasnt able to test it. i think i need some  .env configs to get the app working locally.

Comment (kzl5010) [reply app/src/components/EditScreenInfo.tsx:14]: for the `supabase auth is checked on the webserver side` stuff, i think that should be as straightforward as moving most of these supabase auth methods to the backend

```
 await supabase.auth.signInWithPassword({
      email,
      password,
    });
```

and then making them separate endpoints that the frontend hits and stores the session / data  / result etc

Comment (kzl5010) [reply app/src/components/EditScreenInfo.tsx:14]: that or maybe the backend can check the frontend's auth token with `supabase.auth.getUser`? dunno if that works but would be faster than moving everything to the backend


```
  /**
   * Gets the current user details if there is an existing session.
   * @param jwt Takes in an optional access token jwt. If no jwt is provided, getUser() will attempt to get the jwt from the current session.
   */
  async getUser(jwt?: string): Promise<UserResponse> {
```

File: app/src/api/orders/index.ts (hunk 1 above)
Comment (sourdzl) [reply app/src/api/orders/index.ts:103]: thanks!

**Pull request merged**